trieste/models/interfaces.py

```diff
@@ -287,6 +291,17 @@ def predict_ensemble(self, query_points: TensorType) -> tuple[TensorType, Tensor
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def ensemble_distributions(self, query_points: TensorType) -> tuple[tfd.Distribution, ...]:
```

Comment:
although this resolves typing issues its not great as this is somewhat specific to DeepEnsembles, I wouldn't generally expect ensembles to have individual probabilistic models... @uri-granta perhaps we should dispense with this interface until we get more ensemble models and move the sampler to models.py?